### DISCUSSION TITLE
Mint from specific wallet

### DISCUSSION THREAD

**Discussion by jtsimmons108** [reach-sh/reach-lang#962] [Q&A] [answered]
Hello all,

I'm working on a project where I would like users to be able to build an NFT through a UI and then be able to mint the NFT upon completion.

I found the documentation on how to create a new token at: https://docs.reach.sh/frontend/#launching-tokens 
and I've looked through the other examples on how to then transfer the token to the user that built the NFT.

However, I'm struggling with how to have a safe automated participant in the contract.  I saw in the RPS step 8 tutorial that you can load an account from secret. I would like the Minter Participant to be a specific address, but am unsure how to get the account credentials into the contract in a safe/secure way. I would appreciate any ideas on how to make this happen.  Thanks in advance for your time. 

**Edit**
As I'm continuing to think through this, I think I'm going to try it with an API user to handle the minting on the backend. Would still appreciate feedback if there's a good way to safely do it with an auto participant.

**Comment by jeapostrophe** [comment, marked as answer]
I don't think I understand what you're building, so my advice may be bad.

I think that you want to create an account handle for your user with `stdlib.getDefaultAccount` and it will use their wallet.

If instead you mean that you want to mint on their behalf automatically, then you should absolutely not include any secret in the file that you ship to users. Instead, you'd run a traditional Web2 server that is running a server that is using Reach and it would use `newAccountFromMnemonic`.
